Library/Homebrew/dev-cmd/bump-formula-pr.rb

```diff
@@ -116,6 +117,24 @@ def check_for_duplicate_pull_requests(formula)
   end
 
   def bump_formula_pr
+    @args = Homebrew::CLI::Parser.parse do
+      switch "--devel"
+      switch "-n", "--dry-run"
+      switch "--write"
+      switch "--audit"
+      switch "--strict"
+      switch "--no-browse"
+      switch "--quiet"
+      switch "--force"
+      flag   "--url",      required: true
```

Comment:
We'll need a way to do stuff like this: 
```
(`--url=`<URL> `--sha256=`<sha-256>|`--tag=`<tag> `--revision=`<revision>)
```

i.e. have arguments that are grouped such that if `--url` is specified then `--sha256` is also needed and that only `--url` or `--tag` is allowed/required.

Reply:
Note that is true in principle, but in this command --sha256 is optional even when using --url.

Reply:
So it would need the ability to say `--sha256=` exists, and is optional, if `--url` is passed, but otherwise the option does not exist and is invalid.

Reply:
Thanks @ilovezfs, @MikeMcQuaid : Addressed above comments and added tests for the same.